Add envoy connection balancing. (#14616)
hashicorp/consul: website/content/docs/connect/config-entries/service-defaults.mdx

                      [\`service-intentions\`](/docs/connect/config-entries/service-intentions).
                      Supported values are one of \`tcp\`, \`http\`, \`http2\`, or \`grpc\`.`,
    },
    {
      name: 'BalanceInboundConnections',
      type: `string: ""`,
      description: `Sets the strategy for allocating inbound connections to the service across proxy threads.

>                       The only supported value is \`exact_balance\` for Enovy proxies or undefined
>                       (no connection balancing) behavior. Refer to the

Reviewer: I feel like something like this is slightly more clear:

```suggestion
                      The only supported value is \`exact_balance\`. By default, no connection balancing is used. Refer to the
```
